__tests__/reducers/literal.test.js

```diff
@@ -1,88 +1,136 @@
-import { literal } from '../../src/reducers/literal'
+import { removeAllContent, removeMyItem, setMyItems  } from '../../src/reducers/literal'
 
 describe('literal reducer', () => {
-  it('should handle initial state', () => {
-    expect(
-      literal(undefined, {})
-    ).toEqual({formData: []})
-  })
 
   it('should handle SET_ITEMS', () => {
-    expect(
-      literal({formData: []}, {
-        type: 'SET_ITEMS',
-        payload: {id: 1, uri:'Run the tests', items: []}
-      })
-    ).toEqual({
-      "formData": [{
-        "id": 1, "uri": "Run the tests", "items": []
-      }]
+    const literalSetItems = setMyItems({ "resourceTemplate:Monograph:Instance": {
+      'http://schema.org/name': { items: [] }
+    }}, {
+      type: 'SET_ITEMS',
+      payload: {
+        rtId: 'resourceTemplate:Monograph:Instance',
+        uri:'http://schema.org/name',
+        items: [ { id: 0, content: 'Run the tests'} ]
+      }
+    })
+    expect(literalSetItems).toEqual({
+      "resourceTemplate:Monograph:Instance": {
+        'http://schema.org/name': {
+          items: [{ id: 0, content: 'Run the tests'}]
+        }
+      }
     })
 
     expect(
-      literal({
-        "formData": [{
-          "id": 1, "uri": "Run the tests", "items": []
-        }]}, {
+      setMyItems({ "resourceTemplate:Monograph:Instance": {
+        'http://schema.org/name': {
+          items: [{ id: 1, content: "Run the tests" }] },
+        'http://schema.org/description': {
+          items: []}
+        }
+      },
+      {
         type: 'SET_ITEMS',
-        payload: {id:2, uri: "add this!", items: []}
+        payload: {
+          rtId: "resourceTemplate:Monograph:Instance",
+          uri: 'http://schema.org/description',
+          items: [ { id: 2, content: "add this!"}]
+        }
       })
     ).toEqual({
-      "formData": [
-        {"id": 1, "uri": "Run the tests", "items": []},
-        {"id": 2, "uri": "add this!", "items": []}
-    ]})
+      "resourceTemplate:Monograph:Instance": {
+        'http://schema.org/name': {
+          items: [{ id: 1, content: 'Run the tests'}]
+        },
+        'http://schema.org/description': {
+          items: [{ id: 2, content: "add this!"}]
+        }
+      }
+    })
   })
   it('should handle REMOVE_ITEM', () => {
-    expect(
-      literal({formData: [{id: 1, uri:"Test", items:[
-        {content: "test content", id: 0},
-        {content: "more content", id: 1}
-        ]}]}, {
-        type: 'REMOVE_ITEM',
-        payload: {id: 0, label: "Test"}
-      })
-    ).toEqual({
-      "formData": [{
-        id: 1, uri: "Test", "items": [{content: "more content", id: 1}]
-      }]
+    expect(removeMyItem({
```

Comment:
this seems to show that removing the item in the payload, when it matches one we have, results in that item being removed.  Perhaps the description for this ONE expects could be better?  "REMOVE_ITEM removes the item when it is found" ?

Reply:
Changed description to better reflect the change in state.